sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -98,6 +100,7 @@ class CSVInferSchema(options: CSVOptions) extends Serializable {
           compatibleType(typeSoFar, tryParseDecimal(field)).getOrElse(StringType)
         case DoubleType => tryParseDouble(field)
         case TimestampType => tryParseTimestamp(field)
+        case DateType => tryParseDate(field)
```

Comment:
The problem here is it looks a bit odd that we try date type later. IIRC the root cause is related with date paring library. Couldn't we try date first if we switch the parsing library? I thought that's in progress.

Reply:
Just in case, I did exact match here as well, see https://github.com/apache/spark/pull/23202/files#diff-17719da188b2c15129f848f654a0e6feR174 . If date parser didn't consume all input (`pos.getIndex != field.length`), it fails. If I move it up in type inferring pipeline, it should work.

Reply:
I see. Can we try date first above? was wondering if there was a reason to try date first.

Reply:
Done. Please, have a look at the changes.